test/extended/oauth/client_secret.go

```diff
@@ -107,20 +109,19 @@ var _ = g.Describe("[sig-auth][Feature:OAuthServer]", func() {
 					client := &http.Client{Transport: tr}
 
 					tokenResponse, err = client.Do(tokenReq)
-					// if there was an error then append the error and continue
-					if err != nil {
-						reqErr = append(reqErr, err)
+					// if there is an error or if the response was not 200 then continue
+					if err != nil || tokenResponse.StatusCode != http.StatusOK {
+						framework.Logf("%v", err)
```

Comment:
the information about an unexpected `StatusCode` is lost here, you may want to make these two separate conditions